CMakeLists.txt

```diff
@@ -79,7 +80,7 @@ if (DEFINED PYTHON_INCLUDE_DIR AND DEFINED PYTHON_LIBRARIES AND DEFINED Python_E
     SET(PYTHONLIBS_FOUND TRUE)
     # Use the Python module to find the python lib.
     if(BUILD_WITH_PYTHON_3)
-        find_package(Python 3.7 EXACT REQUIRED COMPONENTS Interpreter)
+        find_package(Python ${BUILD_WITH_PYTHON_3_VERSION} EXACT REQUIRED COMPONENTS Interpreter)
```

Comment:
I reworked my solution as the preflight failed. I realized that you cannot call find_package twice with the same component (in this case Python) because it caches internal variables and actually overwrites the output ones. In my previous case when trying (and failing) to find py 3.9 it would set "Python_EXECUTABLE" to NOT-FOUND. And thus the call to find py 3.7 would also fail. So now I set a cache variable with the py 3 version to find (default is 3.7). Then from our build when we update to 3.9 we can set this.